### DISCUSSION TITLE
What are the bundle size colors (in build output) based on?

### DISCUSSION THREAD

**Discussion by bsmith-self** [vercel/next.js#32301] [Help]
I can see the logic that determines the colors here:
https://github.com/vercel/next.js/blob/eb7b40171a4fcb1f913cad9ef35fa4e2f1a81177/packages/next/build/utils.ts#L107-L115

\>130kb for yellow, >170kb for red. Are those numbers based on any data or official recommendations, or are they arbitrary?

**Comment by wwwdepot** [comment]
Read the reference article[1] about performance budget. 130-170 KB is the budget point where your site loads fast under 5 seconds over 3G connection. 

[1] https://infrequently.org/2017/10/can-you-afford-it-real-world-web-performance-budgets/
